Lib/test/test_tools/test_pathfix.py

```diff
@@ -17,15 +17,17 @@ def setUp(self):
         self.temp_file = support.TESTFN
         self.addCleanup(support.unlink, support.TESTFN)
 
-    def pathfix(self, shebang, pathfix_flags):
+    def pathfix(self, shebang, pathfix_flags, expected_returncode=0, stderr=''):
         with open(self.temp_file, 'w', encoding='utf8') as f:
             f.write(f'{shebang}\n' + 'print("Hello world")\n')
 
         proc = subprocess.run(
             [sys.executable, self.script,
              *pathfix_flags, '-n', self.temp_file],
             capture_output=True)
-        self.assertEqual(proc.returncode, 0, proc)
+
+        self.assertEqual(proc.returncode, expected_returncode, proc)
+        self.assertEqual(proc.stderr.decode(), stderr, proc)
 
         with open(self.temp_file, 'r', encoding='utf8') as f:
             output = f.read()
```

Comment:
If expected_returncode is non-zero, I suggest to check that the input shebang is equal to the output shebang. For example:

```
        new_shebang = lines[0]
        if expected_returncode:
            self.assertEqual(new_shebang, shebang)
        return new_shebang
```